DataContent/DAL/Repositories/UserRepository.cs

```diff
@@ -0,0 +1,92 @@
+﻿using DataContent.DAL.Interfaces;
+using DataContent.DAL.UserExceptions;
+using Microsoft.EntityFrameworkCore;
+using ModelLibrary;
+using ModelLibrary.DataContexts;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Scrypt;
+
+namespace DataContent.DAL.Repositories
+{
+    public class UserRepository : IUserRepository
+    {
+        readonly UserContext _context;
+        public UserRepository(UserContext context)
+        {
+            _context = context;
+        }
+        public async Task<User> DeleteUserAsync(int id)
+        {
+            var user =await _context.Users.Where(x => x.Id == id).FirstOrDefaultAsync();
+            _context.Users.Remove(user);
+            await _context.SaveChangesAsync();
+            return user;
+        }
+
+        public async Task<User> GetUserByIdAsync(int id)
+        {
+            var user = await _context.Users.Where(x => x.Id == id).FirstOrDefaultAsync();
+            return user;
+        }
+
+        public async Task<User> LoginAsync(string email, string password)
+        {
+            var validUser = await _context.Users.Where(x => x.Email == email).FirstOrDefaultAsync();
+            if(validUser == null)
+            {
+                throw new UserNotFoundException("Email is not registered");
+            }
+            ScryptEncoder encoder = new ScryptEncoder();
+            bool isValidUser = encoder.Compare(password, validUser.Password);
+            if (!isValidUser)
+            {
+                throw new WrongUserPasswordException("Password for this user is incorrect");
+            }
+            return validUser;
+        }
+
+        public async Task<User> RegisterAsync(User user)
+        {
+            //check for email in db
+            var registeredUser = await _context.Users.Where(x => x.Email == user.Email).FirstOrDefaultAsync();
+            if(registeredUser != null)
+            {
+                throw new EmailAlreadyRegisteredException("Email is already registered");
+            }
+            //check for username in db
+            registeredUser = await _context.Users.Where(x => x.Username == user.Username).FirstOrDefaultAsync();
+            if(registeredUser != null)
+            {
+                throw new UsernameAlreadyRegisteredException("Username is already used");
```

Comment:
we have talked that in UI we will give different strings to user when he entered wrong password/ email already used, so i think there is no need to create many custom exceptions. Enough to have one with different strings. For example:
RegisterException ("Password for this user is incorrect");
RegisterException (""Email is already registered"");
in this way it will be much easier to handle exceptions and show error messages in UI.

Reply:
Good Idea, haven't thought of that. Done.